### ISSUE TITLE
reset() no longer works

### ISSUE THREAD

**Issue by ribasushi** [Test-More/test-more#549]
The [reset()](https://metacpan.org/pod/Test::Builder#reset) mechanism no longer works as documented. I was unable to come up with a patch, as it requires architectural overview of whether the Stream itself needs to gain a reset method, or whether the T::B reset needs to throw away its stream object and instantiate a new one. 

Consider the difference in output (yes I know this is invalid TAP, see below):

`perl -MTest::More -e 'my $tb = Test::Builder->new; for (1,2,3) { $tb->reset; $tb->ok(1); $tb->ok(2); $tb->done_testing }'`

Beta T::B
```
ok 1
ok 2
1..2
ok 3
ok 4
not ok 5 - done_testing() was already called at -e line 1
#   Failed test 'done_testing() was already called at -e line 1'
#   at -e line 1.
ok 6
ok 7
not ok 8 - done_testing() was already called at -e line 1
#   Failed test 'done_testing() was already called at -e line 1'
#   at -e line 1.
# Looks like you planned 2 tests but ran 8.
# Looks like you failed 2 tests of 8 run.
```

Stable T::B
```
ok 1
ok 2
1..2
ok 1
ok 2
1..2
ok 1
ok 2
1..2
```

While this doesn't make much sense from a practical PoV, it has uses in tests running in persistent environment (just like indicated by the T::B docs themselves). I discovered this when running a [particular DBIC tests](https://github.com/dbsrgits/dbix-class/commit/66917da3#diff-6512506a6e6af2d802755bffacd45d8a) on a system with SpeedyCGI.

**Comment by ribasushi** [comment]
Is there a dev release planned including this? Would be nice to have one at some point so my devrel dep CI jobs can pick it up.

**Comment by exodist** [comment]
yup, I was waiting on feedback on #552, but since that is not gonna happen for a few days I will go ahead and ship a dev release.

**Comment by exodist** [comment]
_099 has been uploaded, should show up on metacpan shortly.
